patches/chrome-browser-profiles-profile_avatar_icon_util.cc.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/chrome/browser/profiles/profile_avatar_icon_util.cc b/chrome/browser/profiles/profile_avatar_icon_util.cc
+index b126966b32db51704639d19c47ed2c1c6c57652a..d58bc5b2196489979c0b1481cd968fd548e6ca9f 100644
+--- a/chrome/browser/profiles/profile_avatar_icon_util.cc
++++ b/chrome/browser/profiles/profile_avatar_icon_util.cc
+@@ -261,7 +261,7 @@ constexpr char kHighResAvatarFolderName[] = "Avatars";
+ 
+ // The size of the function-static kDefaultAvatarIconResources array below.
+ #if !defined(OS_CHROMEOS) && !defined(OS_ANDROID)
+-constexpr size_t kDefaultAvatarIconsCount = 56;
++constexpr size_t kDefaultAvatarIconsCount = 56 + kBraveDefaultAvatarIconsCount;
+ #else
+ constexpr size_t kDefaultAvatarIconsCount = 27;
+ #endif
+@@ -362,6 +362,7 @@ size_t GetPlaceholderAvatarIndex() {
+ 
+ size_t GetModernAvatarIconStartIndex() {
+ #if !defined(OS_CHROMEOS) && !defined(OS_ANDROID)
++  BRAVE_GET_AVATAR_ICON_START_INDEX
+   return GetPlaceholderAvatarIndex() + 1;
+ #else
+   // Only use the placeholder avatar on ChromeOS and Android.
+@@ -387,6 +388,7 @@ std::string GetPlaceholderAvatarIconUrl() {
+ 
+ const IconResourceInfo* GetDefaultAvatarIconResourceInfo(size_t index) {
+   CHECK_LT(index, kDefaultAvatarIconsCount);
++  BRAVE_RETURN_IF_BRAVE_DEFAULT_AVATAR_RESOURCE_INFO
+   static const IconResourceInfo resource_info[kDefaultAvatarIconsCount] = {
```

Comment:
The interesting side effect here is that chromium's resource_info will be sized bigger than it needs to be. There can't be an issue here since we handle any index that is greater than `kDefaultAvatarIconsCount - kBraveDefaultAvatarIconsCount` in the brave function.
We cannot insert the items below since this is defined as `const`, and if we try and `#define` the brave items in the middle of the array definition syntax (if that's possible) then we'll not get the `static_assert` that @darkdh rightly suggested we include.
I think it's fine as it is, but happy to take suggestions.